src/main/java/io/cdap/plugin/cloud/vision/transform/transformer/FaceAnnotationsToRecordTransformer.java

```diff
@@ -0,0 +1,175 @@
+                  annotation.getLandmarkingConfidence());
+    }
+    if (faceSchema.getField(ImageExtractorConstants.FaceAnnotation.ANGER_FIELD_NAME) != null) {
+      builder.set(ImageExtractorConstants.FaceAnnotation.ANGER_FIELD_NAME, annotation.getAngerLikelihood().name());
+    }
+    if (faceSchema.getField(ImageExtractorConstants.FaceAnnotation.JOY_FIELD_NAME) != null) {
+      builder.set(ImageExtractorConstants.FaceAnnotation.JOY_FIELD_NAME, annotation.getJoyLikelihood().name());
+    }
+    if (faceSchema.getField(ImageExtractorConstants.FaceAnnotation.BLURRED_FIELD_NAME) != null) {
+      builder.set(ImageExtractorConstants.FaceAnnotation.BLURRED_FIELD_NAME, annotation.getBlurredLikelihood().name());
+    }
+    if (faceSchema.getField(ImageExtractorConstants.FaceAnnotation.SORROW_FIELD_NAME) != null) {
+      builder.set(ImageExtractorConstants.FaceAnnotation.SORROW_FIELD_NAME, annotation.getSorrowLikelihood().name());
+    }
+    if (faceSchema.getField(ImageExtractorConstants.FaceAnnotation.UNDER_EXPOSED_FIELD_NAME) != null) {
+      builder.set(ImageExtractorConstants.FaceAnnotation.UNDER_EXPOSED_FIELD_NAME,
+                  annotation.getUnderExposedLikelihood().name());
+    }
+    if (faceSchema.getField(ImageExtractorConstants.FaceAnnotation.HEADWEAR_FIELD_NAME) != null) {
+      builder.set(ImageExtractorConstants.FaceAnnotation.HEADWEAR_FIELD_NAME,
+                  annotation.getHeadwearLikelihood().name());
+    }
+    if (faceSchema.getField(ImageExtractorConstants.FaceAnnotation.SURPRISE_FIELD_NAME) != null) {
+      String surprise = annotation.getSurpriseLikelihood().name();
+      builder.set(ImageExtractorConstants.FaceAnnotation.SURPRISE_FIELD_NAME, surprise);
+    }
+    Schema.Field positionField = faceSchema.getField(ImageExtractorConstants.FaceAnnotation.POSITION_FIELD_NAME);
+    if (positionField != null) {
+      Schema positionArraySchema = positionField.getSchema().isNullable() ? positionField.getSchema().getNonNullable()
+        : positionField.getSchema();
+      Schema positionSchema = positionArraySchema.getComponentSchema().isNullable()
+        ? positionArraySchema.getComponentSchema().getNonNullable()
+        : positionArraySchema.getComponentSchema();
+
+      List<StructuredRecord> position = annotation.getBoundingPoly().getVerticesList().stream()
+        .map(v -> extractVertex(v, positionSchema))
+        .collect(Collectors.toList());
+      builder.set(ImageExtractorConstants.FaceAnnotation.POSITION_FIELD_NAME, position);
+    }
+    Schema.Field fdPositionField = faceSchema.getField(ImageExtractorConstants.FaceAnnotation.FD_POSITION_FIELD_NAME);
+    if (fdPositionField != null) {
+      Schema positionArraySchema = fdPositionField.getSchema().isNullable()
+        ? fdPositionField.getSchema().getNonNullable()
+        : fdPositionField.getSchema();
+      Schema positionSchema = positionArraySchema.getComponentSchema().isNullable()
```

Comment:
don't you think it should be a helper method somewhere to avoid this extraction everywhere?

Reply:
A method that extracts a non-nullable component's schema of the specified array field added to the base class.
```
  /**
   * Extracts non-nullable component's schema of the specified array field.
   *
   * @param field array field.
   * @return non-nullable component's schema of the specified array field
   */
  protected Schema getComponentSchema(Schema.Field field) {
    Schema arraySchema = field.getSchema().isNullable() ? field.getSchema().getNonNullable() : field.getSchema();
    Schema componentSchema = Objects.requireNonNull(arraySchema.getComponentSchema());
    return componentSchema.isNullable() ? componentSchema.getNonNullable() : componentSchema;
  }
```